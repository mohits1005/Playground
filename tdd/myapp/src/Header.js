import React from 'react';
function Header({ text }) {
    return (
        <header>
            <h1 className='fancy-h1' data-testid='h1tag'>
                {text}
            </h1>
        </header>
    );
}
export default Header;