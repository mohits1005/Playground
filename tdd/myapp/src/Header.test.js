import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import Header from './Header';

afterEach(cleanup);

it('render', () => {
    const { asFragment } = render(<Header text="Header" />);
    //snapshot test
    expect(asFragment()).toMatchSnapshot();
})

it('inserts text in h1', () => {
    const { getByTestId, getByText } = render(<Header text='Header' />);
    expect(getByTestId('h1tag')).toHaveTextContent('Header');
    expect(getByText('Header')).toHaveClass('fancy-h1');
})
