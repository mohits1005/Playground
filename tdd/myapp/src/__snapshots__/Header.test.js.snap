// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="fancy-h1"
      data-testid="h1tag"
    >
      Header
    </h1>
  </header>
</DocumentFragment>
`;
